src/tests/params.rs

```diff
@@ -9,6 +9,38 @@ mk_tests_default! {
     use chrono::{NaiveDate, NaiveTime};
     use rand::{distributions::Standard, Rng};
 
+    #[cfg(feature = "linking")]
+    #[cfg(not(feature = "embedded_tests"))]
+    #[test]
+    fn charsets() -> Result<(), FbError> {
+        use crate::{ConnectionBuilder, charset::ISO_8859_1};
+
+        let mut conn = ConnectionBuilder::linked()
+            .charset(ISO_8859_1)
+            .connect()
+            .expect("Error on connect the test database using the ISO8859_1");
+
+        conn.execute("DROP TABLE pCHARSETS", ()).ok();
+        conn.execute(
+            "CREATE TABLE pCHARSETS (a Varchar(30) CHARACTER SET none)",
+            (),
+        )?;
+
+        conn.execute("insert into pCHARSETS (a) values (?)", ("Pão de queijo",))?;
+
+        let (pao,): (String,) = conn.query_first("select * from pCHARSETS", ())?
+            .unwrap();
+        assert_eq!("Pão de queijo", pao);
+
+        let mut conn = connect(); // utf8
+
+        let err: Result<Option<(String,)>, FbError> = conn.query_first("select * from PCHARSETS", ());
+        assert!(err.is_err());
+        assert_eq!("error: Found column with an invalid UTF-8 string: invalid utf-8 sequence of 1 bytes from index 1", err.err().unwrap().to_string());
+
+        Ok(())
+    }
```

Comment:
The charset tests only runs in the `linking` version. I thing it's better to crate a new module for the charset tests and return two connections, one `UTF8 ` encoded and one `ISO_8859_1` encoded using the `mk_tests!` macro.

Reply:
A new module for charset OK, but I did not understand the "return thow connections". You are talking about the 'connect()' function?

Reply:
Yes, you can change it to return a tuple with two connections for example

Reply:
Creating a copy of the macro defined in line 44 of the `tests/mod.rs` file, that returns two connections instead of one, to use in the charset tests, that was my idea.

Reply:
Now I get it, but i think that we can use a other approach. Instead of duplicate the mk_tests_default, we can return a ConnectionBuilder instance. This will provide a greater flexibility for ours tests.